src/net/client/protocol.rs

```diff
@@ -98,23 +98,21 @@ impl<Addr> TlsConnect<Addr> {
     }
 }
 
-impl<Addr> AsyncConnect for TlsConnect<Addr>
-where
-    Addr: ToSocketAddrs + Clone + Send + 'static,
-{
+impl AsyncConnect for TlsConnect {
```

Comment:
This change prevents TcpStream::connect() from attempting to connect in turn to each of multiple addresses until one is successful. Is that intentional?